app/src/main/java/com/amaze/filemanager/ui/fragments/data/MainFragmentViewModel.kt

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2014-2021 Arpit Khurana <arpitkh96@gmail.com>, Vishal Nehra <vishalmeham2@gmail.com>,
+ * Emmanuel Messulam<emmanuelbendavid@gmail.com>, Raymond Lai <airwave209gt at gmail.com> and Contributors.
+ *
+ * This file is part of Amaze File Manager.
+ *
+ * Amaze File Manager is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.amaze.filemanager.ui.fragments.data
+
+import android.content.SharedPreferences
+import android.os.Bundle
+import androidx.lifecycle.ViewModel
+import com.amaze.filemanager.adapters.data.LayoutElementParcelable
+import com.amaze.filemanager.database.CloudHandler
+import com.amaze.filemanager.file_operations.filesystem.OpenMode
+import com.amaze.filemanager.filesystem.HybridFileParcelable
+import com.amaze.filemanager.ui.fragments.preference_fragments.PreferencesConstants
+import com.amaze.filemanager.utils.DataUtils
+import java.util.*
+
+class MainFragmentViewModel : ViewModel() {
+
+    var currentPath: String? = null
+
+    /** This is not an exact copy of the elements in the adapter  */
+    var listElements: ArrayList<LayoutElementParcelable>? = null
```

Comment:
This should never be null.

Reply:
It was initialized as null in previous implementation in MainFragment. Not taking any risk to change any logic for now.

Reply:
An issue should be created to fix this. Will create.